analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/LockConstraint.cs

```diff
@@ -0,0 +1,36 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Constraints
+{
+    public sealed class LockConstraint : SymbolicConstraint
+    {
+        public static readonly LockConstraint Held = new();
```

Comment:
Maybe wee need more than just held or released in order to be able to track if the reader or the writer lock is held/released?

Reply:
Held/Released should be enough for S2222. 
Tracking `Read/Write` kind of lock will be another (new) rule. See the google docs/FIXMEs/last point.

Reply:
I missed that comment.